### DISCUSSION TITLE
Can I force the use of HTTP APIs instead of gRPC? Can these libraries be installed where there is no gRPC support?

### DISCUSSION THREAD

**Discussion by Indavelopers** [googleapis/google-cloud-python#12687] [General]
Hi all, thanks for checking this issue.
I'm trying to use several GC Client Libraries for Python in a different environment, [PyScript](https://pyscript.com/).

PyScript uses Pyodide and WebAssembly to bring Python to the browser client-side.

In order to use PyScript/Pyodide in the browser, you need a pure Python wheel, or build one yourself. The problem is Pyodide uses a different distribution and architecture, for web assembly, installed via micropip ([more info](https://pyodide.org/en/stable/usage/loading-packages.html)).

I'm working to build pure Python wheels for web assembly ([docs](https://pyodide.org/en/stable/development/building-and-testing-packages.html)), but I'm finding myself at a stop.

Most of GC Client Libraries for Python list gRPC as a requirement, e.g. sometimes as `google-api-core[grpc]`.

I believe I've read that these libraries fall back to HTTP if gRPC is not available, but my problem is that, if I'm not mistaken, they try to install grpcio library locally, which throws an error in PyScript/Pyodide/Web Assembly environments, as there is no grpcio library available for this architecture, nor seems to be possible as for now to execute gRPC calls from the browser (at least using grpcio).

Therefore, my questions are:

1. Do these libraries support environments without gRPC support?
1. Is there a way for setting up these Python Cloud Client Libraries for only HTTP support, without the need to install grpcio?
1. How is local gRPC support checked, if any? Could we mock a local gRPC library for the installing error not to arise, but the fallback to correctly use HTTP?

Looking forward to ideas and responses, thanks!

**Comment by parthea** [comment]
1. Yes, most of the Google Cloud Libraries support both gRPC and REST transport. As an example, you can use the following code to set the transport to gRPC for the `google-cloud-access-approval` client.   
In this [Sample code](https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-access-approval/samples/generated_samples/accessapproval_v1_generated_access_approval_list_approval_requests_sync.py
) change
```
    # Create a client
    client = accessapproval_v1.AccessApprovalClient()
```
to
```
    # Create a client
    client = accessapproval_v1.AccessApprovalClient(transport="rest")
```
2. You can install a library without dependencies by running `pip install google-cloud-access-approval --no-deps`. Then you can selectively choose which dependencies you prefer. We cannot remove the dependency on gRPC in the default installation as gRPC is the default transport. Note, you may run into issues by not having gRPC installed, so please file issues and we will address them

3. There is no need to mock gRPC. Please file issues if you encounter a problem without gRPC installed and we will address it.
